src/App.tsx

```diff
@@ -1,61 +1,200 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
-
 import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
+import { User } from './types/User';
+import { getUsers } from './api/Users';
+import { getPosts } from './api/Posts';
+import { Post } from './types/Post';
 import { Loader } from './components/Loader';
+import { Comment } from './types/Comment';
+import { addComment, deleteComment, getComments } from './api/Comments';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isLoader, setLoader] = useState(false);
+  // const [selectedPost, setSelectedPost] = useState<Post>();
+  const [selectedPostId, setSelectedPostId] = useState(0);
+  const [comments, setComments] = useState<Comment []>([]);
+  const [postLoadingError, setPostLoadingError] = useState(false);
+  const [commentsLoadingError, setCommentsLoadingError] = useState(false);
+  const [commentAddError, setCommentAddError] = useState(false);
+  const [commentDeleteError, setCommentDeleteError] = useState(false);
+  const [isCommentsLoader, setCommentsLoader] = useState(false);
+  const [isFormVisible, setFormVisability] = useState(false);
+  const [isCommentAdding, setCommentAdding] = useState(false);
+
+  const isPostListVisible = Boolean(!isLoader
+    && posts.length && selectedUserId);
+
+  const hasNoPosts = Boolean(!isLoader && selectedUserId && !posts.length);
+
+  const selectedPost = posts.find((post) => post.id === selectedPostId);
+
+  useEffect(() => {
+    getUsers().then(setUsers);
+  }, []);
+
+  const handleSelectedUser = async (userId: number) => {
+    setSelectedUserId(userId);
+    setLoader(true);
+    setSelectedPostId(0);
+
+    try {
+      const postsFromServer = await getPosts(userId);
```

Comment:
let's be consistent:
you use "wait" and "then" in this component, let's use one